pkg/controllers/vdb/obj_reconciler.go

```diff
@@ -272,6 +272,16 @@ func (o *ObjReconciler) checkForDeletedSubcluster(ctx context.Context) (ctrl.Res
 		if err != nil {
 			return ctrl.Result{}, err
 		}
+		// Delete vproxy config map
+		if vmeta.UseVProxy(o.Vdb.Annotations) {
+			scName := stss.Items[i].Labels[vmeta.SubclusterNameLabel]
```

Comment:
you should check that sts contains the subcluster label.

Reply:
Sure, added validation in https://github.com/vertica/vertica-kubernetes/pull/980/commits/ec75345ef8c24a0d98579ef71e4996f440a966ef